### PR TITLE
Fix escaping of apostrophes and newlines in inbox

### LINKED ISSUE #535: Contacting the reporter about a ticket of a game with an apostrophe in the title
When clicking on "Contact the reporter", the title of the PM is cut where the apostrophe is, and the heading of the message doesn't appear.

### PR DESCRIPTION
**Pull request by Jamiras**
Fixes newlines in messages being converted to "rn" and apostrophes prematurely truncating message titles (#535).

image

I chose to reimplement #536 and #537 as I believe a better solution is to url encode the parameter (instead of c escaping it) and then always htmlencode the string when building the new page (instead of only encoding the value read from the database and expecting the url parameter to be encoded).

Closes #536
Closes #537 
Fixes #535

### CODE REVIEW

File: lib/database/message.php [hunk 1]
```diff
@@ -2,10 +2,10 @@
 
 function CreateNewMessage($author, $destUser, $messageTitle, $messagePayloadIn)
 {
-    sanitize_sql_inputs($author, $destUser, $messageTitle, $messagePayloadIn);
-
     $messagePayload = nl2br($messagePayloadIn);
 
+    sanitize_sql_inputs($author, $destUser, $messageTitle, $messagePayloadIn);
```
Comment (Jamiras) [inline lib/database/message.php:7]: the important change here is actually moving the `messagePayload` line, so the <br> conversion is performed on a not-sanitized input.

File: lib/database/ticket.php [hunk 2]
```diff
@@ -114,10 +114,10 @@ function submitNewTickets($userSubmitter, $idsCSV, $reportType, $noteIn, &$summa
         return false;
     }
 
-    sanitize_sql_inputs($userSubmitter, $reportType, $noteIn);
+    $note = $noteIn;
+    sanitize_sql_inputs($userSubmitter, $reportType, $note);
```
Comment (Jamiras) [inline lib/database/ticket.php:118]: sanitize a copy of the message so the unescaped value can be passed to later functions.

File: lib/database/message.php (hunk 1 above)
Comment (luchaos) [reply lib/database/message.php:7]: makes sense 👍

**Review by luchaos — APPROVED** [review]
lgtm, tested locally.

**Pull request merged**